### PR TITLE
[enhancement] Add catch_signal test utility

### LINKED ISSUE #87: Add catch_signal test utility
- add [catch_signal from openwisp-monitoring](https://github.com/openwisp/openwisp-monitoring/blob/e3dfc323c72b7c29d6ecf7ff126dcee96da51df7/openwisp_monitoring/tests.py#L1-L11) to openwisp-utils
- add at least 1 test for it
- add a few lines of docs
- update [openwisp-controller ](https://github.com/openwisp/openwisp-controller/pull/179/files)and [openwisp-monitoring](https://github.com/openwisp/openwisp-monitoring/blob/e3dfc323c72b7c29d6ecf7ff126dcee96da51df7/openwisp_monitoring/tests.py#L1-L11) to use that and remove the duplication

### PR DESCRIPTION
**Pull request by nepython**
closes #87

### CODE REVIEW

**Comment by nepython** [comment]
@nemesisdesign, I thought it was appropriate to move the `catch_signal` to `utils.py` though I am unsure. If it needs to be relocated please do suggest :smiley:

**Comment by nepython** [comment]
>add a bullet point here: https://github.com/openwisp/openwisp-utils#current-features, mention the test utilities, link the section you added before

@nemesisdesign, should I add a link, since the `Table of Contents` right above already has the link and none of the other features in `Current Features` have links so I have not yet added it just to maintain consistency.
>**Please reference the issue you're working on in the commit message.**

Oops, just noticed that all my commits recently didn't refer one. My bad.

PS: I might not be good at documenting as haven't not done much documentation so please feel free to suggest any edits though I have tried to keep it simple and provided use case for the same.

**Comment by nepython** [comment]
**Update**: Implemented it successfully across **openwisp-controller** :smile:

File: README.rst [hunk 1]
```diff
@@ -319,10 +320,32 @@ Admin mixins
...
+
+Usage example as a context-manager::
+
+    with catch_signal(status_signal) as handler:
+            <Perform the action that generates the signal>
+        handler.assert_called_once_with(
+            <parameters>=<arguments>,
+            sender=<Sender name>,
+            signal=<Signal name>,
+        )
```
Comment (TheOneAboveAllTitan) [inline README.rst:343]: Please define code block language as done in [here](https://github.com/openwisp/openwisp-utils/pull/88/files#diff-88b99bb28683bd5b7e3a204826ead112R309). 

Making example copy paste friendly may help future contributors who are working with this feature for first time. What do you say @nepython ?
Also comment out helper strings as
```
<Perform the action that generates the signal>
``

Comment (nepython) [reply README.rst:343]: >Making example copy paste friendly may help future contributors who are working with this feature for first time. What do you say @nepython ?

Actually @TheOneAboveAllTitan, I have copy pasted the actual usage of this feature just replacing **specific names of signals and parameters** with help-text in **<>** as I found it as a general practice in openwisp and elsewhere. I am new to documenting changes, open to all suggestions to make it as user-friendly as possible:smiley:. Thanks for the suggestions.

Comment (TheOneAboveAllTitan) [reply README.rst]: Sorry, I made a typo in the previous comment. :sweat: 
But you got it right anyway :+1: :smile:

**Pull request merged**